openforcefield/tests/test_molecule.py

```diff
@@ -2882,3 +2887,87 @@ def test_visualize_openeye(self):
         mol = Molecule().from_smiles("CCO")
 
         assert isinstance(mol.visualize(backend="openeye"), IPython.core.display.Image)
+
+
+class MyMol(FrozenMolecule):
+    """
+    Lightweight FrozenMolecule subclass for molecule-subclass tests below
+    """
+
```

Comment:
I don't know if it consequential that this test class inherits from `FrozenMolecule` vs. `Molecule`; I can't immediately think of a way to break it.

```
In [12]: class MyOtherMol(Molecule):
    ...:     pass
    ...:

In [13]: isinstance(MyOtherMol.from_smiles("CCO"), Molecule)
Out[13]: True

In [14]: isinstance(MyOtherMol.from_smiles("CCO"), FrozenMolecule)
Out[14]: True
```

Reply:
It's a bit consequential since `FrozenMolecule` doesn't expose `add_atom`, but instead only exposes the private `_add_atom` method (same with `_add_bond` and `_add_conformer`). So I think it's better this way, since using `FrozenMolecule` for these tests actually makes them stricter than just vanilla `Molecule`.